nss/integration-tests/nss_test.go

```diff
@@ -53,30 +53,29 @@ func TestIntegration(t *testing.T) {
-		"Returns empty when getting all entries from passwd and daemon is not available": {db: "passwd", noDaemon: true},
-		"Returns empty when getting all entries from group and daemon is not available":  {db: "group", noDaemon: true},
-		"Returns empty when getting all entries from shadow and daemon is not available": {db: "shadow", noDaemon: true},
+		// We can't assert on the returned error type since the error returned by getent will always be 2 (i.e. Not Found), even though the library returns other types.
+		"Error when getting passwd by name and entry does not exist": {db: "passwd", key: "doesnotexit", wantErr: true},
+		"Error when getting group by name and entry does not exist":  {db: "group", key: "doesnotexit", wantErr: true},
+		"Error when getting shadow by name and entry does not exist": {db: "shadow", key: "doesnotexit", wantErr: true},
 
-		"Returns empty when getting all entries from passwd and database is corrupted": {db: "passwd", cacheDB: "invalid_entry_in_userByID"},
-		"Returns empty when getting all entries from group and database is corrupted":  {db: "group", cacheDB: "invalid_entry_in_groupByID"},
-		"Returns empty when getting all entries from shadow and database is corrupted": {db: "shadow", cacheDB: "invalid_entry_in_userByID"},
+		"Error when getting passwd by id and entry does not exist": {db: "passwd", key: "404", wantErr: true},
+		"Error when getting group by id and entry does not exist":  {db: "group", key: "404", wantErr: true},
 
-		// We can't assert on the returned error type since the error returned by getent will always be 2 (i.e. Not Found), even though the library returns other types.
 		"Error when getting passwd by name and daemon is not available": {db: "passwd", key: "user1", noDaemon: true, wantErr: true},
 		"Error when getting group by name and daemon is not available":  {db: "group", key: "group1", noDaemon: true, wantErr: true},
 		"Error when getting shadow by name and daemon is not available": {db: "shadow", key: "user1", noDaemon: true, wantErr: true},
 
 		"Error when getting passwd by id and daemon is not available": {db: "passwd", key: "1111", noDaemon: true, wantErr: true},
 		"Error when getting group by id and daemon is not available":  {db: "group", key: "11111", noDaemon: true, wantErr: true},
 
-		"Error when getting passwd by name and entry does not exist": {db: "passwd", key: "doesnotexit", wantErr: true},
-		"Error when getting group by name and entry does not exist":  {db: "group", key: "doesnotexit", wantErr: true},
-		"Error when getting shadow by name and entry does not exist": {db: "shadow", key: "doesnotexit", wantErr: true},
+		// Even though those are "error" cases, the getent command won't fail since the other databases on the machine will return some entries.
+		"Returns empty when getting all entries from passwd and daemon is not available": {db: "passwd", noDaemon: true},
```

Comment:
they are coming last while I asked them to come before the errors. This is always what you acked in your comment: "You mean reordering the cases so the "entry does not exist" ones come first?"

Reply:
Yeah, I misunderstood the comment... I thought you still wanted these to be considered errors and just wanted a reordering on the error ones. I'm sorry about that.